windows/devices/uwb/include/windows/devices/uwb/UwbDeviceConnector.hxx

```diff
@@ -192,8 +195,23 @@ private:
     void
     OnSessionRangingData(::uwb::protocol::fira::UwbRangingData rangingData);
 
+    /**
+     * @brief Internal function to check if there are callbacks present
+     *
+     * @return true
+     * @return false
+     */
+    bool
+    CallbacksPresent();
+
 private:
+    mutable std::shared_mutex m_eventCallbacksGate;
+
+    mutable std::mutex m_callbacksPresentConditionVariableGate;

```

Comment:
2 separate mutexes aren't necessary since they're both protecting the same state. You can use the `std::shared_mutex` here as long as you change `m_callbacksPresentConditionVariable` to be a `std::condition_variable_any` and ensure that the lock you pass to `wait()` is a `std::unique_lock` of the `m_eventCallbacksGate`. When `wait()` is called, the mutex is atomically _released_, making it available to synchronize access to `m_sessionEventCallbacks` and `m_deviceEventCallbacks` in whatever way you want (ie. either with `std::shared_lock`, `std::unique_lock`, `std::scoped_lock`, etc.)